src/db/SQLJSPackageDB.ts

```diff
@@ -105,17 +104,23 @@ const INSERT_RESOURCE = `INSERT INTO resource
 const SD_FLAVORS = ['Extension', 'Logical', 'Profile', 'Resource', 'Type'];
 
 export class SQLJSPackageDB implements PackageDB {
+  private db: Database;
   private insertPackageStmt: Statement;
   private insertResourceStmt: Statement;
   private findAllPackagesStmt: Statement;
   private findPackagesStmt: Statement;
   private findPackageStmt: Statement;
   private optimized: boolean;
-  constructor(
-    private db: Database,
-    initialize = true
-  ) {
-    if (initialize) {
+  private initialized: boolean;
+
+  constructor() {
+    this.initialized = false;
+  }
+
+  async initialize() {
+    const SQL = await initSqlJs();
+    if (!this.initialized) {
```

Comment:
I thought about this. Here's the thing: `initSqlJs` is asynchronous. So... if we have this:
```ts
if (!this.initialized) {
  const SQL = await initSqlJs();
  // a bunch of other code
  this.initialized = true;
}
```
... then it is possible that while we are awaiting the initSqlJs, _another_ code path can come into the condition since `this.initialized` is still false (since the await allows the thread to continue processing other code paths while it waits). This is _highly unlikely_, but technically possible. Since I don't expect the initialization state to be checked often, I thought it was probably OK to put the async call before the `if` statement to avoid the potential race condition.

There are other options though...
1. Put `initSqlJs` inside the block but set `this.initialized = true` before invoking it. The danger there is that while we await `initSqlJs`, if another codepath gets here, it will think SQLJSPackageDB is already initialized when it's actually not yet initialized. So that's not very good.
2. Put `initSqlJs` inside the block and _after_ it returns, check `this.initialized` _again_ before doing the real initialization. I guess that maybe that is the best way, as it still protects against the race condition but also minimizes cases in which `initSqlJs` is called multiple times.

So maybe I should change it to do `#2` above?

Reply:
I went ahead and took the approach described above in point 2. See: 813c629.

Reply:
I like the approach in point 2. But I also understand now the original approach (point 0 perhaps). I probably would have agreed that your original approach was sufficient and slightly simpler, but now that you updated it, I like that approach too!